### PR TITLE
TSV source for langNames #115 

### PR DESCRIPTION
**Pull request by ayushjainrksh**
I have started working on the issue 115 and created tsv files for sql queries. I haven't deleted the original sql files for now. I have also written python script that needs to be build at `make` so that tsv files may be converted to sql and inserted to db.

### CODE REVIEW

**Comment by sushain97** [comment]
Please remove all the extra commits. You should be working on a branch without them and merging into master here.

**Comment by ayushjainrksh** [comment]
Okay I will remove all the extra commits. Please check my PRs.

**Comment by sushain97** [comment]
I can't review this PR until the extra commits are reviewed. The changeset is a mess.

**Comment by ayushjainrksh** [comment]
I have removed most of the extra commits.

**Comment by sushain97** [comment]
I'd still like the other commits to be removed. Please use a branch so development is simple. You can cherry-pick the relevant commits into it.

Also, I think the TSVs don't need the number at the start. They can be autogenerated?

**Comment by ayushjainrksh** [comment]
I have removed the ids at the beginning of the tsv files and they are auto-generating now. Should I use a single branch for all the PRs I send? If so how do I manage the different issues?

**Comment by sushain97** [comment]
No, each PR should come from a different branch on your fork.

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Requirements
 ------------
 
 - Python 3.4+
-- Tornado 4.3+ (`python3-tornado` on Debian)
+- Tornado 4.3-5.1.1 (`python3-tornado` on Debian)
```
Comment (sushain97) [inline README.md]: Why are these changes here? Maybe try merging master into your branch.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <title>Apertium APy</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+</head>
+<body>
+    <h1>Apertium APy</h1>
+    <h3><a href="http://wiki.apertium.org/wiki/Apertium-apy">Documentation</a></h3>
+    <h3><a href="https://github.com/apertium/apertium-apy">Source</a></h3>
+</body>
+</html>
```
Comment (sushain97) [inline index.html]: I think you need to merge origin master into your branch.

File: language_names/additions.tsv [hunk 3]
```diff
@@ -0,0 +1,149 @@
+en	lvs	Latvian
```
Comment (sushain97) [inline language_names/additions.tsv]: Is it possible to use a TSV with headers? I'm pretty sure the Python lib supports it. This would be much more readable that way.

File: language_names/manual.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import sqlite3
+import textwrap
+import csv
+
+
+def populate_database():
+    conn = sqlite3.connect('test.db')
+    c = conn.cursor()
+    c.execute(textwrap.dedent("""
```
Comment (sushain97) [inline language_names/manual.py]: This SQL statement and the one below doesn't really need dedent.

File: language_names/manual.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        PRAGMA foreign_keys=OFF;"""))
+    c.execute(textwrap.dedent("""
+        BEGIN TRANSACTION;"""))
+    c.execute(textwrap.dedent("""
+        CREATE TABLE IF NOT EXISTS fixes (id integer primary key, lg text, inLg text, name text, unique(lg, inLg) on conflict replace);
```
Comment (sushain97) [inline language_names/manual.py]: Suggest each field on its own line here and below. Also, ON CONFLICT REPLACE should be capital as should field types, e.g. INTEGER PRIMARY KEY.

File: language_names/manual.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        """))
+    with open('fixes.tsv', 'r') as f:
```
Comment (sushain97) [inline language_names/manual.py]: This loop would probably be best extracted into a function since you do the same thing twice here and you'll have to do it again (for the large SQL file).

File: README.md (hunk 1 above)
Comment (ayushjainrksh) [reply README.md]: Okay I will do that

File: language_names/additions.tsv (hunk 3 above)
Comment (ayushjainrksh) [reply language_names/additions.tsv]: Yeah it can be done

File: language_names/manual.py (hunk 4 above)
Comment (ayushjainrksh) [reply language_names/manual.py]: okay I will verify

File: language_names/manual.py (hunk 5 above)
Comment (ayushjainrksh) [reply language_names/manual.py]: Okay I'll make these changes

File: language_names/manual.py (hunk 6 above)
Comment (ayushjainrksh) [reply language_names/manual.py]: Okay I have written a function for it.

File: index.html (hunk 2 above)
Comment (ayushjainrksh) [reply index.html]: I have merged origin master now, sorry for this.

File: apertium_apy/apy.py [hunk 7]
```diff
@@ -178,11 +184,17 @@ def apply_config(args, parser, apy_section):
...
+        lang_names_path = os.path.join(os.path.dirname(apy_path), lang_names_default_path)
+    else:
+        lang_names_path = lang_names_default_path
     parser = argparse.ArgumentParser(description='Apertium APY -- API server for machine translation and language analysis')
     parser.add_argument('pairs_path', help='path to Apertium installed pairs (all modes files in this path are included)')
     parser.add_argument('-s', '--nonpairs-path', help='path to Apertium tree (only non-translator debug modes are included from this path)')
     parser.add_argument('-l', '--lang-names',
-                        help='path to localised language names sqlite database (default = langNames.db)', default='langNames.db')
+                        help='path to localised language names sqlite database (default = %s)' % (lang_names_path),
+                        default=lang_names_path)
```
Comment (sushain97) [inline apertium_apy/apy.py]: These changes still shouldn't be here. I think your branch has some extra commits it shouldn't.

File: language_names/additions.tsv [hunk 8]
```diff
@@ -0,0 +1,150 @@
+lg  inLg    name
```
Comment (sushain97) [inline language_names/additions.tsv]: This has spaces as separators, not tabs.

File: language_names/fixes.tsv [hunk 9]
```diff
@@ -0,0 +1,272 @@
+lg  inLg    name
```
Comment (sushain97) [inline language_names/fixes.tsv]: This has spaces as separators, not tabs.

File: language_names/manual.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+def insert_values(c, filename):
+    c.execute(textwrap.dedent("""
+        CREATE TABLE IF NOT EXISTS {file} (
+            id INTEGER PRIMARY KEY,
+            lg TEXT,
+            inLg TEXT,
+            name TEXT,
+            UNIQUE(lg, inLg) ON CONFLICT REPLACE);
+        """.format(file=filename)))
```
Comment (sushain97) [inline language_names/manual.py]: Is there a near to use keyword arguments here? Just a `{}` in the string should suffice.

File: language_names/manual.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    with open(filename + '.tsv', 'r') as f:
```
Comment (sushain97) [inline language_names/manual.py]: Please use string formatting here.

File: language_names/manual.py [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+        reader = csv.reader(f, delimiter='\t')
```
Comment (sushain97) [inline language_names/manual.py]: Using a `DictReader` will allow avoiding the weird `next(...)` below.

File: language_names/manual.py [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        next(reader)
+        for row in reader:
+            c.execute('INSERT INTO ' + filename + ' VALUES (?, ?, ?, ?)', (None, row[0], row[1], row[2]))
+
+
+def populate_database():
+    conn = sqlite3.connect('langNames.db')
+    c = conn.cursor()
+    c.execute("""
```
Comment (sushain97) [inline language_names/manual.py]: This statement and the one below don't need to be multiple lines.

File: language_names/manual.py (hunk 10 above)
Comment (ayushjainrksh) [reply language_names/manual.py]: Okay it can be done using string interpolation

File: language_names/manual.py (hunk 12 above)
Comment (ayushjainrksh) [reply language_names/manual.py]: yeah that would be great

File: language_names/manual.py (hunk 11 above)
Comment (ayushjainrksh) [reply language_names/manual.py]: Okay that can be done.

File: language_names/manual.py (hunk 13 above)
Comment (ayushjainrksh) [reply language_names/manual.py]: I will change it.
